app/src/main/java/com/eveningoutpost/dexdrip/webservices/XdripWebService.java

```diff
@@ -213,13 +218,25 @@ private void handle(Socket socket) throws IOException {
             // Read HTTP headers and parse out the route.
             reader = new BufferedReader(new InputStreamReader(socket.getInputStream()));
             String line;
+            boolean authFailed = false;
+
             while (!TextUtils.isEmpty(line = reader.readLine())) {
+                if (line.startsWith(("api-secret"))) {
+                    String secret = Pref.getStringDefaultBlank("xdrip_webservice_secret");
+                    if (!secret.equals("")) {
+                        secret = sha1Hash(secret);
```

Comment:
Use something like ` final String hashedSecret = Hashing.sha1().hashBytes(secret.getBytes(Charsets.UTF_8)).toString();` instead to avoid duplicating code?